### PR TITLE
Initial implementation of task start, list, and delete commands; adde…

### PR DESCRIPTION
**Pull request by agodwinrelay**
…d placeholders for rest of capsule commands

We want to merge the capsule CLI commands into the relay CLI, so that we are able to start/schedule tasks, create task types, etc through this CLI and eventually do away with the temporary CLI we have for capsule. This branch will eventually contain all of the new commands from the capsule CLI. Right now, I've only added commands for starting, listing, and deleting a task (account.ts, system.ts, list.ts, and delete.ts). I wanted to get some feedback from you before I continue on to make sure I'm following the right pattern that we use for all of the existing commands in the CLI, so this PR doesn't have to actually be merged yet.

### CODE REVIEW

File: src/commands/capsule/tasks/start/system.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+const debug = debugFn(`capsule:tasks:start:system`)
+
+export default class TasksStartSystemCommand extends Command {
+
+  static description = `Start a system task with the given configuration`
+  static hidden = true
+
+  static flags = { // how to change ordering of these in help command?
+    task_type_name: string({
```
Comment (BrandonSmith) [inline src/commands/capsule/tasks/start/system.ts:16]: Flags should not have underscores `_` in the name as it makes it awkward to type: `--task_type_name`

Further, as much as possible, strive to reduce typing an unnecessary context in flag names. For instance, `--task_type_name` in the context of a task command can simply be `--type`.

I'd rename, and be consistent across all task commands, to the following:

--type
--major or --version
--name
--assign-to
--file
--tag

File: src/commands/capsule/tasks/start/system.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  async run(): Promise<void> {
+    const { flags } = await this.parse(TasksStartSystemCommand)
+    var encoded_string = flags.args
+    if (encoded_string.charAt(0) == '@') {
+      var encoded_string = fs.readFileSync(encoded_string.substring(1, encoded_string.length),{ encoding: 'utf8', flag: 'r' }).toString()
+    }
+    flags.args = JSON.parse(encoded_string)
+
+    if (flags.tag) {
+        flags.args = appendTag(flags.args, flags.tag)
```
Comment (BrandonSmith) [inline src/commands/capsule/tasks/start/system.ts:63]: I'd like to better understand what the value of `flags.args.tags` should be. Is it a string delimited by commas or is it an array of strings?

Do we need to merge tag values from the JSON file or do we overwrite the tags value if `--tag` is provided?

This is a common merge pattern:
```
flags.args.tags = {
  ...flags.args,
  tags: [ ...flags.args.tags, tag ]
}
```

Comment (agodwinrelay) [reply src/commands/capsule/tasks/start/system.ts:63]: It should be an array of strings (which I am just now realizing should by default always contain the task type name along with the specified tag). For example, when starting a delivery task in the capsule CLI:
```
$ ./capsule tasks start system delivery 1 test frog @delivery_args.json --tag myTag
```
You get (see tags in args):
```
$ ./capsule tasks list
[{"assign_to":["urn:relay-resource:name:device:frog"],"task_name":"test","subscriber_id":"1234","task_type_name":"delivery","status":"running","task_type_namespace":"system","args":{"stoppable_code":"myCode1234","tags":["myTag","delivery"],"sla":{"max_deviation":60,"target_from_start":80},"deliverers":["frog","sloth"],"task_timeout":200,"min_time_in_destination":60,"verbose":true,"destination":["tower a","1st","cafeteria"]},"task_id":"56IkWenwj9AfrXrhvnRN9A0A","workflow_id":"wf_56IkWenwj9AfrXrhvnRN9A0A_P9A25yGMmVNgPt6asjJnqoA","timestamp":"2023-09-21T19:14:11Z","workflow_instance_id":"JV2nDhAKZiQhOjqRLZsATC","task_type_major":1}]

```

**Comment by agodwinrelay** [comment]
> The relay-cli project is using ESLint to strive to have consistency across code. So there are a lot of similar warnings. Mostly about whitespace (number spaces per tab) and use of backtick for strings, except where JS doesn't allow. There are a few others in there. These will need to be fixed.
> 
> GitHub is nice enough to run through ESLint during the PR process. I can help show you how to see these warnings and errors before committing and submitting a PR.
> 
> Next, I'd like to sync on some of these commands to better understand the hierarchy. It is ok if you are following along with the Lua implementation. But I'd like to take this time to figure out how it should fit into the CLI. For instance...
> 
> What is the the `relay capsule workflow` commands going to do that is different then the existing `relay workflow`?
> 
> I wonder if we should remove `capsule` from the CLI command hierarchy and utilize `task` as the root? Resulting in...
> 
> ```
> relay task [events | start | delete | list | schedule]
> relay task type [create | delete | list | update]
> ```
> 
> I don't think we should have duplicative code for each namespace of account and system, etc. I think this can be handled as a flag `--namespace` (with a default of `account`?). And perhaps the flag is even hidden so only internal users know about it?...
> 
> ```
> // instead of
> relay task start account
> // it would be start command shared with..
> relay task start --namespace account
> ```
> 
> More comments in line with code.
> 
> Sorry if some of this is free flowing thoughts. :)

Sounds good! And yeah, definitely missed all of those ESLint errors after making the PR..

Marcel and I weren't sure how I should go about implementing `workflow` within the CLI, and were planning on having discussions on what that should look like. 

I like the idea of removing `capsule` from the command hierarchy as I find myself forgetting to specify that half the time. Also,  will add in the `--namespace` flag, it definitely felt slightly hacky how I had it but wanted to try and get something working first :sweat_smile: . 

Thanks for the feedback! We should definitely take some time to sync on how some of these should look in the relay CLI as opposed to their current implementation in the capsule CLI.

**Review by BrandonSmith — APPROVED** [review]

**Pull request merged**